scripts/third_party_size_check_test.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for scripts/third_party_size_check.py"""
+
+from __future__ import annotations
+
+import builtins
+import os
+import shutil
+import sys
+
+from core import utils
+from core.tests import test_utils
+from scripts import third_party_size_check
+
+from typing import Any
+
+
+class ThirdPartySizeCheckTests(test_utils.GenericTestBase):
+    """Unit tests for scripts/third_party_size_check.py"""
+
+    def test_get_skip_files_list(self) -> None:
+        skip_files_list = (
+            'random_file.py\n'
+            '# This is a comment\n'
+            'new_file.py')
+        with open('dummy_skip_file_list.txt', 'w', encoding='utf-8') as f:
+            f.write(skip_files_list)
+
+        dummy_file_object = open(
+            'dummy_skip_file_list.txt', 'r', encoding='utf-8')
+
+        swap_open = self.swap_with_checks(
+            utils, 'open_file',
+            lambda *unused_args, **unused_kwargs: dummy_file_object,
+            expected_args=(('.gcloudignore', 'r'),))
+
+        with swap_open:
+            returned_list = third_party_size_check._get_skip_files_list() # pylint: disable=protected-access
+
+        skipped_files = ['random_file.py', 'new_file.py']
+        expected_skipped_files_list = [
+            os.path.join(os.getcwd(), file) for file in skipped_files]
+        print(expected_skipped_files_list)
+        self.assertEqual(returned_list, expected_skipped_files_list)
+
+        dummy_file_object.close()
+        os.remove('dummy_skip_file_list.txt')
```

Comment:
If the test fails, we want to make sure this dummy file gets cleaned up. You can use `tempfile` to create temporary files that will automatically get cleaned up by the operating system. Ditto elsewhere

Reply:
`tempfile` deals in binary inputs (like the **write()** method accepts bytes inputs while **read()** method outputs bytes which creates consistency issues with the code. Moreover, `tempfile` cannot be used to create other UNIX filetypes needed to completely test the file (as referred to in a comment above).
To overcome this, I have moved all the file/directory creation code in `setUp()` method while the code to remove these dummy files and dirs in `teardown()` method which executes after every test is run.

Reply:
OK, I think your current approach is fine. Just for the future though, tempfile.TemporaryFile accepts a `mode` argument that you can use to control whether binary input is expected, and you can use tempfile.TemporaryDirectory to create a directory within which you can create any kind of file you like